core/src/test/scala/unit/kafka/log/OffsetIndexTest.scala

```diff
@@ -178,6 +178,14 @@ class OffsetIndexTest extends JUnitSuite {
     // mmap should be null after unmap causing lookup to throw a NPE
     intercept[NullPointerException](idx.lookup(1))
   }
+
+  @Test
+  def testSanityWithSingleMessage(): Unit = {
```

Comment:
nit: it's not just that it's a single message, but that the offset of the one message matches the segment base offset.